### ISSUE TITLE
False positive in Style/HashEachMethods

### ISSUE THREAD

**Issue by dteoh** [rubocop-hq/rubocop#7831]
I have code like this:

```ruby
def validate_values
  # values is a Hash
  values.each do |key, value|
    next if value.is_a? String

    errors.add :values, :unsupported, key: key
  end
end
```
--------

## Expected behavior

No error is raised by Style/HashEachMethods.

## Actual behavior

```
 C: Style/HashEachMethods: Use each_value instead of values.each.
    values.each do |key, value|
    ^^^^^^^^^^^
```

## Steps to reproduce the problem

Run RuboCop on the code snippet.

## RuboCop version

```
⟩ bundle exec rubocop -V
0.80.1 (using Parser 2.7.0.5, running on ruby 2.5.5 x86_64-darwin19)
```
